beacon-chain/sync/rpc_beacon_blocks_by_root.go

```diff
@@ -46,11 +47,40 @@ func (s *Service) sendBeaconBlocksRequest(ctx context.Context, requests *types.B
 		}
 		return nil
 	})
+
+	log := log.WithField("peer", id)
+
+	blkByRoot := make(map[[fieldparams.RootLength]byte]interfaces.ReadOnlySignedBeaconBlock)
 	for _, blk := range blks {
+		blkRoot, err := blk.Block().HashTreeRoot()
+		if err != nil {
+			return errors.Wrap(err, "block hash tree root")
+		}
+
+		blkByRoot[blkRoot] = blk
+	}
+
+	if err != nil {
```

Comment:
Isn't this error check for `SendBeaconBlocksByRootRequest` so shouldn't it be on line 50 and the `log := log.WithField("peer", id)` above it?

Reply:
I agree with you. My first attempt was to add an error check after calling `SendBeaconBlocksByRootRequest`, and return early. But it starting making some tests to fail.

Let's ask @terencechain why he chose this design.

Reply:
Aligned with @saolyn here, also if there is an error and `blks` variable is nil then it doesn't make sense to have `for _, blk := range blks` thing before checking error.